### PR TITLE
[Merged by Bors] - feat: port Data.Fintype.Sum

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fintype/Sum.lean [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  rfl
+#align finset.univ_disj_sum_univ Finset.univ_disjSum_univ
+
+@[simp]
+theorem Fintype.card_sum [Fintype α] [Fintype β] :
+    Fintype.card (Sum α β) = Fintype.card α + Fintype.card β :=
+  card_disjSum _ _
+#align fintype.card_sum Fintype.card_sum
+
+/-- If the subtype of all-but-one elements is a `fintype` then the type itself is a `fintype`. -/
```
Comment (ChrisHughes24) [inline Mathlib/Data/Fintype/Sum.lean:44]:
```suggestion
/-- If the subtype of all-but-one elements is a `Fintype` then the type itself is a `Fintype`. -/
```

File: Mathlib/Data/Fintype/Sum.lean [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  card_disjSum _ _
+#align fintype.card_sum Fintype.card_sum
+
+/-- If the subtype of all-but-one elements is a `Fintype` then the type itself is a `Fintype`. -/
+def fintypeOfFintypeNe (a : α) (h : Fintype { b // b ≠ a }) : Fintype α :=
+  Fintype.ofBijective (Sum.elim ((↑) : { b // b = a } → α) ((↑) : { b // b ≠ a } → α)) <| by
+    classical exact (Equiv.sumCompl (· = a)).bijective
+#align fintype_of_fintype_ne fintypeOfFintypeNe
+
+-- Porting note: `image` picked up the classical instance automatically in lean3.
```
Comment (jcommelin) [inline Mathlib/Data/Fintype/Sum.lean:50]: Does `open Classical in` help?

**Comment by jcommelin** [comment]
Apart from the comment about Classical, LGTM

**Comment by jcommelin** [comment]
bors d+

**Pull request closed without merging**